src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosResponseProcessor.cs

```diff
@@ -113,6 +118,16 @@ public async Task ProcessResponseAsync(CosmosResponseMessage responseMessage, Ca
                 double.TryParse(responseMessage.Headers["x-ms-request-duration-ms"], out var duration) ? duration : 0,
                 responseMessage.Headers["x-ms-documentdb-partitionkeyrangeid"]);
 
+            if (_cosmosDataStoreConfiguration.LogSDKDiagnostics)
+            {
+                _queryLogger.LogQueryDiagnostics(responseMessage.Headers.ActivityId, responseMessage.Diagnostics);
+            }
+
+            if (_cosmosDataStoreConfiguration.LogSDKClientElapsedTime)
+            {
+                _queryLogger.LogQueryClientElapsedTime(responseMessage.Headers.ActivityId, responseMessage.Diagnostics?.GetClientElapsedTime().ToString());
+            }
```

Comment:
If this is null should we to the check in the if?

```suggestion
            if (_cosmosDataStoreConfiguration.LogSDKClientElapsedTime && responseMessage.Diagnostics != null)
            {
                _queryLogger.LogQueryClientElapsedTime(responseMessage.Headers.ActivityId, responseMessage.Diagnostics?.GetClientElapsedTime().ToString());
            }
```

Reply:
Fixed